<html>
<head>
  <title>Evernote Export</title>
  <basefont face="Segoe UI" size="2" />
  <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  <meta name="exporter-version" content="Evernote Windows/308292 (en-US, MWS); Windows/10.0.0 (Win64);"/>
  <style>
    body, td {
      font-family: Segoe UI;
      font-size: 10pt;
    }
  </style>
</head>
<body>
<a name="390"/>

<div><span><div><b>EXAMPLE OF GENE ID: </b></div><div>Resource from: <a href="https://www.encodeproject.org/targets/2L52.1-celegans/">https://www.encodeproject.org/targets/2L52.1-celegans/</a></div><div>Target name
</div><dd>2L52.1
</dd><dt>Target gene
</dt><ul><li><div><a href="https://www.ncbi.nlm.nih.gov/gene/181792">GeneID:181792</a></div></li></ul><dt>External resources
</dt><ul><li><div><a href="http://www.wormbase.org/species/c_elegans/gene/WBGene00007063">WormBase:WBGene00007063</a></div></li><li><div><a href="http://www.wormbase.org/species/c_elegans/gene/2L52.1">WormBase:2L52.1</a></div></li><li><div><a href="https://www.ncbi.nlm.nih.gov/gene/181792">GeneID:181792</a></div></li><li><div><a href="http://www.uniprot.org/uniprot/A4F336">UniProtKB:A4F336</a></div></li></ul><div> </div><div>Notes: </div><ul><li><div>coexpresssion.csv is in Target name</div></li><li><div><br/></div></li></ul><div><br/></div></span>
</div>
<hr>
<a name="391"/>

<div><span><div>File: testgff.sh </div><ul><li><div>Prints all on chromosome 1 </div></li><li><div>LOCATION: workspace/248</div></li><li><div>STATUS: success </div></li><li><div>Learned from: <a href="https://www.biostars.org/p/200279/">https://www.biostars.org/p/200279/</a></div></li></ul><div><br/></div><div><br/></div></span>
</div>
<hr>
<a name="392"/>

<div>
<span><div>Followed this tutorial: </div><div><a href="http://meme-suite.org/doc/install.html?man_type=web#quick">http://meme-suite.org/doc/install.html?man_type=web#quick</a></div><div>“Quick install” </div><div><br/></div><div>Where to find /etc</div><div><a href="http://www.dowdandassociates.com/blog/content/howto-set-an-environment-variable-in-mac-os-x-terminal-only/">http://www.dowdandassociates.com/blog/content/howto-set-an-environment-variable-in-mac-os-x-terminal-only/</a></div><div><br/></div><div>Reminder: Safety copy files to desktop first with</div><div><br/></div><div>cp -p profile /Users/rwoo/Desktop</div><div><br/></div><div>DID NOT WORK RIGHT AWAY: </div><div>Edited file: </div><div><a href="https://groups.google.com/forum/#!searchin/meme-suite/terminal%7Csort:date">https://groups.google.com/forum/#!searchin/meme-suite/terminal%7Csort:date</a></div><div><br/></div><div>REDO INSTALL MEME ON NEW COMPUTER:<br/>
Need to unpack in cygwin </div><div>tar xzvf &lt;.tar.gz file&gt;</div><div><br/></div><div>Learning about MEME: <a href="http://meme-suite.org/doc/meme.html">http://meme-suite.org/doc/meme.html</a></div><div><br/></div><div>Programming help: </div><div><a href="https://wiki.haskell.org/How_to_unpack_a_tar_file_in_Windows">https://wiki.haskell.org/How_to_unpack_a_tar_file_in_Windows</a></div></span>
</div>
<hr>
<a name="393"/>

<div><span><div>GFF file format: </div><div><a href="https://useast.ensembl.org/info/website/upload/gff.html">https://useast.ensembl.org/info/website/upload/gff.html</a></div><div><br/></div><ol><li><div><span style="font-weight: bold;">seqname</span> - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. <span style="font-weight: bold;">Important note</span>: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.</div></li><li><div><span style="font-weight: bold;">source</span> - name of the program that generated this feature, or the data source (database or project name)</div></li><li><div><span style="font-weight: bold;">feature</span> - feature type name, e.g. Gene, Variation, Similarity</div></li><li><div><span style="font-weight: bold;">start</span> - Start position of the feature, with sequence numbering starting at 1.</div></li><li><div><span style="font-weight: bold;">end</span> - End position of the feature, with sequence numbering starting at 1.</div></li><li><div><span style="font-weight: bold;">score</span> - A floating point value.</div></li><li><div><span style="font-weight: bold;">strand</span> - defined as + (forward) or - (reverse).</div></li><li><div><span style="font-weight: bold;">frame</span> - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..</div></li><li><div><span style="font-weight: bold;">attribute</span> - A semicolon-separated list of tag-value pairs, providing additional information about each feature.</div></li></ol><div></div><div><br/></div><div>Basic awk usage: </div><div><a href="https://alvinalexander.com/blog/post/linux-unix/any-easy-way-print-column-or-columns-from-text-file-using-awk">https://alvinalexander.com/blog/post/linux-unix/any-easy-way-print-column-or-columns-from-text-file-using-awk</a></div><div><br/></div><div>Example Usage: </div><ul><li><div>GET ALL WITH CHR1 </div></li><li><div>awk '{ print $1=&quot;CHROMOSOME_I&quot; }' 50.gff2</div></li><li><div>awk '{ print $2=&quot;Genomic_canonical&quot; }' 50.gff2</div></li><li><div>grep -n  &quot;Y74C9&quot; 50.gff2</div></li></ul><div><br/></div><div>Which do I use? </div><ul><li><div>awk '{ print $2=&quot;Genomic_canonical&quot;, $14 }' 50.gff2</div></li><li><div>awk '{ print $2=&quot;Genomic_canonical”, $9!= “ &quot; $10 }' 50.gff2</div></li></ul><div><br/></div><div>
<span style="font: 12.0px Times; font-kerning: none; color: #000000; -webkit-text-stroke: 0px #000000;">CDS: &quot;A contiguous sequence which begins with, and includes, a start codon and ends with, and includes, a stop codon.&quot;</span></div><div><span style="font: 12.0px Times; font-kerning: none; color: #000000; -webkit-text-stroke: 0px #000000;">
</span></div><div><br/></div></span>
</div>
<hr>
<a name="395"/>

<div><span><div><b><u>COMMANDLINE: </u></b></div><div><b><u><br/></u></b></div><div><u>MYSQL: </u><br/></div><div><br/></div><div><span style="font-weight: bold;">SQL review and installation </span></div><ol><li><div><a href="https://gist.github.com/nrollr/3f57fc15ded7dddddcc4e82fe137b58e">https://gist.github.com/nrollr/3f57fc15ded7dddddcc4e82fe137b58e</a></div></li><li><div><a href="https://www.a2hosting.com/kb/developer-corner/mysql/managing-mysql-databases-and-users-from-the-command-line">https://www.a2hosting.com/kb/developer-corner/mysql/managing-mysql-databases-and-users-from-the-command-line</a></div></li><li><div><a href="http://www.cisred.org/content/databases_methods/cele_4/data_files/">http://www.cisred.org/content/databases_methods/cele_4/data_files/</a></div></li></ol><div><br/></div><div>IMPORTANT: </div><div>MYSQL PASSWORD is: ‘mysql’</div><div>User: rachel </div><div>Password: mysql</div><div><br/></div><div><u>BEDTOOLS2:</u></div><div><b><u><br/></u></b></div><div><u>AWK: </u></div><div><u><br/></u></div></span>
</div>
<hr>
<a name="396"/>

<div><span><div>Use 248 version from previous cutter et al. analysis </div><div><br/></div><div><span style="font-weight: bold; text-decoration: underline;">WORMBASE ROOT DIRECTORY FILES: </span></div><div><a href="ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/">ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/</a></div><div><br/></div><div><span style="font-weight: bold;">Genomic Sequence (intergenic) 248: </span></div><div><a href="ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/sequence/genomic/">ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/sequence/genomic/</a></div><ul><li><div>/Users/Desktop/workspace/248/c_elegans.PRJNA13758.WS248.intergenic_sequences.fa</div></li></ul><div><br/></div><div><span style="font-weight: bold;">GFF file: </span></div><div><a href="ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/gff/">ftp://ftp.wormbase.org/pub/wormbase/species/c_elegans/gff/</a></div><ul><li><div>/Users/Desktop/workspace/248/c_elegans.PRJNA13758.WS248.annotations.gff2</div></li></ul><div><br/></div><div><span style="font-weight: bold; text-decoration: underline;">TFBS List: </span></div><div>Methods from this Study: </div><div><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2651782/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2651782/</a></div><div><br/></div><div>Sleumer M. C., M. Bilenky, A. He, G. Roberson, N. Thiessen and S. J. M. Jones. 2009. Caenorhabditis elegant cisRED a catalogue of conserved genomic elements. Nucleic Acid Res. 37(4): 1323-1334.</div><div><br/></div><div>Tool available at: </div><div><a href="http://www.cisred.org/">http://www.cisred.org/</a></div><div> </div><div>This study used data from: </div><ol><li><div>Bryne JC, Valen E, Tang ME, Marstrand T, Winther O, da Piedade I, Krogh A, Lenhard B, Sandelin A. JASPAR, the open access database of transcription factor-binding profiles: new content and tools in the 2008 update. Nucleic Acids Res. 2008;36:102–106.</div></li><li><div>Matys V, Kel-Margoulis OV, Fricke E, Liebich I, Land S, Barre-Dirrie A, Reuter I, Chekmenev D, Krull M, Hornischer K, et al. TRANSFAC and its module TRANSCompel: transcriptional gene regulation in eukaryotes. Nucleic Acids Res. 2006;34:108–110.</div></li><li><div>Griffith OL, Montgomery SB, Bernier B, Chu B, Kasaian K, Aerts S, Mahony S, Sleumer MC, Bilenky M, Haeussler M, et al. ORegAnno: an open-access community-driven resource for regulatory annotation. Nucleic Acids Res. 2008;36:D107–D113.</div></li></ol><div><br/></div><ul><li><div><a href="http://www.cisred.org/content/databases_methods/cele_4/data_files/">http://www.cisred.org/content/databases_methods/cele_4/data_files/</a></div></li></ul><div>NOTE: IN SQL, see above for instructions </div><div><br/></div><div><b><br/></b></div><div><br/></div><div><br/></div></span>
</div>
<hr>
<a name="398"/>

<div><span><ol><li><div>Use R package for scripting</div></li></ol><ul><li><div>Users/Desktop/workspace</div></li><li><div>In r package </div></li><li><div>See: <a href="http://r-pkgs.had.co.nz/">http://r-pkgs.had.co.nz/</a></div></li></ul><div><br/></div><ol start="2"><li><div>Use Terminal commands</div></li></ol><ul><li><div>Users/Desktop/workspace/terminal</div></li><li><div>Only plain text file</div></li></ul><div><span><span><span><br/></span></span></span></div><div><span><span><span><br/></span></span></span></div><div><span><span><span><br/></span></span></span></div></span>
</div>
<hr>
<a name="399"/>

<div><span><div><span style="font-size: 18px; font-family: Arial;">Overview of Wormbase: </span></div><div><a href="https://www.wormbase.org/about/userguide/nomenclature#badfmi19g68klc32eh75j40--10" style="font-size: 18px; font-family: Arial;">https://www.wormbase.org/about/userguide/nomenclature#badfmi19g68klc32eh75j40--10</a></div><div style="margin-top: 1em; margin-bottom: 1em;"><span style="-en-paragraph:true;">The chromosomes of </span><span style="font-style: italic;-en-paragraph:true;">C. elegans</span><span style="-en-paragraph:true;"> have the names:
</span></div><ul><li><div>CHROMOSOME_I</div></li><li><div>CHROMOSOME_II</div></li><li><div>CHROMOSOME_III</div></li><li><div>CHROMOSOME_IV</div></li><li><div>CHROMOSOME_V</div></li><li><div>CHROMOSOME_X</div></li><li><div>CHROMOSOME_MtDNA</div></li></ul><div><br/></div><div>Overview of Worm Anatomy: </div><div><a href="http://www.wormatlas.org/hermaphrodite/introduction/mainframe.htm">http://www.wormatlas.org/hermaphrodite/introduction/mainframe.htm</a></div><div></div><div><font face="Arial" style="font-size: 18px;"><br/></font></div><div><font face="Arial" style="font-size: 18px;">Good Article for Gene Expression Comodules?</font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(0, 0, 0);">Functional Annotation of Caenorhabditis elegans Genes by Analysis of Gene Co-Expression Networks (has tagged but not very specific) </span></font></div><ul><li><div><span style="font-size: 18px; font-family: Arial;">In downloaded </span></div></li></ul><div><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(0, 0, 0);"><font face="Arial" style="font-size: 18px;"><br/></font></span></div><div><span style="font-stretch: normal; line-height: normal;"><font face="Arial"><span style="font-size: 18px;">Worm Life Stages for C. Elegans </span></font></span></div><div><a href="https://wormbase.org/species/all/life_stage/WBls:0000002#012--10">https://wormbase.org/species/all/life_stage/WBls:0000002#012--10</a></div><div><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(0, 0, 0);"><font face="Arial" style="font-size: 18px;"><br/></font></span></div><div><span style="font-family: Helvetica; font-size: 18px;">Worm Life Stages according to Wormbase: </span></div><h2>L1 larva Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000024
</div><div>The first stage larva. At 25 Centigrade, it ranges 14-25.5 hours after fertilization, 0-11.5 hours after hatch.</div><div><br/></div><h2>L1-L2 ecdysis Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000042
</div><div>The stage during L1-L2 molt, after the brief lethargus period, when an animal sheds off old cuticle.
</div><div><br/></div><h2>L1-L2 lethargus Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000025
</div><div>The stage during L1-L2 molt when an animal becomes lethargic, ceases pharyngeal pumping and synthesizes new cuticle under the old.
</div><div><br/></div><h2>L1-L2 molt Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000026
</div><div>The stage when an animal shifts from L1 larva to L2 larva. It includes the synthesis of new cuticle, cease of phrayngeal pumping during a lethargus stage, and the shed off of old cuticle.
</div><div><br/></div><h2>L2 larva Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000027
</div><div>The second stage larva. At 25 Centigrade, it ranges 25.5-32.5 hours after fertilization, 11.5-18.5 hours after hatch.
</div><div><br/></div><h2>L2-L3 ecdysis Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000047
</div><div>The stage during L2-L3 molt, after the brief lethargus period, when an animal sheds off old cuticle.
</div><div><br/></div><h2>L2-L3 lethargus Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000028
</div><div>The stage during L2-L3 molt when an animal becomes lethargic, ceases pharyngeal pumping and synthesizes new cuticle under the old.
</div><div><br/></div><h2>L2-L3 molt Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000029
</div><div>The stage when an animal shifts from L2 larva to L3 larva. It includes the synthesis of new cuticle, cease of phrayngeal pumping during a lethargus stage, and the shed off of old cuticle.
</div><div><br/></div><h2>L3 larva Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000035
</div><div>The third stage larva. At 25 Centigrade, it ranges 32.5-40 hours after fertilization, 18.5-26 hours after hatch.
</div><div><br/></div><h2>L3-L4 ecdysis Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000049
</div><div>The stage during L3-L4 molt, after the brief lethargus period, when an animal sheds off old cuticle.
</div><div><br/></div><h2>L3-L4 lethargus Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000036
</div><div>The stage during L3-L4 molt when an animal becomes lethargic, ceases pharyngeal pumping and synthesizes new cuticle under the old.
</div><div><br/></div><h2>L3-L4 molt Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000037
</div><div>The stage when an animal shifts from L3 larva to L4 larva. It includes the synthesis of new cuticle, cease of phrayngeal pumping during a lethargus stage, and the shed off of old cuticle.
</div><div><br/></div><h2>L4 larva Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000038
</div><div>The fourth stage larva. At 25 Centigrade, it ranges 40-49.5 hours after fertilization, 26-35.5 hours after hatch.
</div><div><br/></div><h2>L4-adult ecdysis
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000050
</div><div>The stage during L4-adult molt, after the brief lethargus period, when an animal shed off old cuticle.
</div><div><br/></div><h2>L4-adult lethargus Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000039
</div><div>The stage during L4-adult molt when an animal becomes lethargic, ceases pharyngeal pumping and synthesizes new cuticle under the old.
</div><div><br/></div><h2>L4-adult molt Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000040
</div><div>The stage when an animal shifts from L4 larva to adult. It includes the synthesis of new cuticle, cease of phrayngeal pumping during a lethargus stage, and the shed off of old cuticle.
</div><div><br/></div><h2>adult Ce
</h2><div><span title="">WormBase ID:</span></div><div>WBls:0000041
</div><div>The stage after an animal is fully-developed and reaches maturity.
</div><div><br/></div><div><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(0, 0, 0);"><font face="Arial" style="font-size: 18px;">THIS TIME STAGE IS FROM LEVIN AND YANI</font></span></div><div>
<font style="font-size: 18px; font-family: Arial;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">4 cell:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">E div:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">4</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">th</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">ABdiv:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">5</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">th</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">ABdiv:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">6</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">th</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">ABdiv:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">7</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">th</span><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">ABdiv:</span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">VE: </span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">CS: </span></font></div><div><font face="Arial" style="font-size: 18px;"><span style="font-style: normal; font-variant-caps: normal; font-weight: normal; font-stretch: normal; line-height: normal; color: rgb(14, 14, 13);">Mov:</span></font></div><div><font face="Arial" style="font-size: 18px;">L1: </font></div><div><font face="Arial" style="font-size: 18px;">L2:</font></div><div><font face="Arial" style="font-size: 18px;">L3: </font></div><div><font face="Arial" style="font-size: 18px;">L4: </font></div><div><span style="font-size: 18px; font-family: Helvetica;"><br/></span></div><div><span style="font-size: 18px; font-family: Helvetica;">Not entirely sure which of these stages correspond to WB stages. </span></div><div><span style="font-size: 18px; font-family: Helvetica;"><br/></span></div><div><span style="font-size: 18px; font-family: Helvetica;">ACTION: DOWNLOADED ALL </span></div></span>
</div>
<hr>
<a name="432"/>

<div>
<span><div><div><ol><ol><li><div>Parse coexpression into 11 different modules</div></li></ol></ol><div><br/></div><div>cd &quot;\Users\rwoo\Desktop\BCB330\Converting Names</div></div><div><br/></div><div>NEW COMPUTER PROTOCOL:</div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">#go to relevant directory</span></div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">cd &quot;/cygdrive/c/Users/rwoo/Desktop/BCB330/Converting Names&quot;</span></div><div><br/></div><div>#only keep GeneID and Module Columns</div><div>cut -d &quot;,&quot; -f1,5 coexpression.csv &gt; outfile.csv</div><div><br/></div><div>#Sort by 14 Module, each new one will be sorted in 1.csv, 2.csv etc. where the number is the coexpression module</div><div>for i in {0..13} do</div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">awk -F , '$&quot;$i&quot; == &quot;$i&quot; { print }' outfile.csv &gt; $ i.csv</span></div><div><span style="min-height: 9pt; font-size: 9pt;"><br/></span></div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">#if done individually: </span><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;"> </span></div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">awk -F , '$1 == &quot;1&quot; { print }' outfile.csv &gt; 1.csv</span></div><div><span style="font-size: 9pt;"><br/></span></div><div><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">#rename GeneID to WormbaseID</span></div><ul><li><div><span style="font-size: 9pt; font-family: &quot;Lucida Console&quot;; color: rgb(1, 1, 1);">used wormmine online </span></div></li><li><div><span style="font-size: 9pt;"><a href="http://intermine.wormbase.org/tools/wormmine/buildBag.do" style="font-family: &quot;Lucida Console&quot;; font-size: 9pt;">http://intermine.wormbase.org/tools/wormmine/buildBag.do</a></span></div></li><li><div><span style="font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">Could use resources here:</span></div></li><li><div><span style="font-size: 9pt;"><a href="https://github.com/WormBase/website/issues/4297" style="font-family: &quot;Lucida Console&quot;; font-size: 9pt;">https://github.com/WormBase/website/issues/4297</a></span></div></li></ul><div><br/></div><div>References (programming): </div><div><a href="https://stackoverflow.com/questions/1560393/bash-shell-scripting-csv-parsing">https://stackoverflow.com/questions/1560393/bash-shell-scripting-csv-parsing</a></div><div><a href="https://stackoverflow.com/questions/18642188/sorting-csv-file-by-5th-column-using-bash">https://stackoverflow.com/questions/18642188/sorting-csv-file-by-5th-column-using-bash</a></div><div><br/></div><div>***** IMPORTANT MISSING NAMES ARE IN CONVERTINGNAMES</div><div><br/></div><div>Wormbase genes are in &quot;converting genes extra info&quot;</div><div><br/></div><div>Troubleshooting commands:</div><div>awk -F &quot;\&quot;*,\&quot;*&quot; '{print $2}' 1wb.csv</div><div><br/></div><div><br/></div><div><br/></div></div><div><br/></div></span>
</div>
<hr>
<a name="454"/>

<div><span><div>Running MEME 1.0</div><div><br/></div><div>Ran online, only got 3 motifs each. Not very useful. Redo on computer. </div><div><br/></div><div>Running MEME 1.1</div><div><br/></div><div>Ran on unused laptop (Timed out at 5) </div><div><br/></div><div>Running MEME 1.2</div><div><br/></div><div>Ran</div><div><br/></div><div>Running MEME 2.0</div><div><br/></div><div>cd /Users/rwoo/Desktop/330/1</div><div><span style="font-size: 12px; -webkit-text-stroke-width: 0px; -webkit-text-stroke-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Courier; font-stretch: normal; font-style: normal; font-variant-caps: normal; font-weight: normal; line-height: normal;">meme 1.txt -dna -oc . -nostatus -time 60000 -mod anr -nmotifs 15 -minw 6 -maxw 25 -objfun de -neg 1control.txt -revcomp -markov_order 1</span></div><div><br/></div><div><span style="font-size: 12px; -webkit-text-stroke-width: 0px; -webkit-text-stroke-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: Courier; font-stretch: normal; font-variant-caps: normal; line-height: normal;">meme 14.txt -dna -oc . -nostatus -time 60000 -mod anr -nmotifs 15 -minw 6 -maxw 25 -objfun de -neg 14control.txt -revcomp -markov_order 1</span></div><div><br/></div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;"> sed -n 2541,2550p 11.txt</span></div><div><br/></div><div><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">from 11</span></div><div style="min-height: 9pt; text-align: left;"><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">&gt;caenorhabditis_elegans_prjna13758|WBGene00014450</span></div><div style="min-height: 9pt; text-align: left;"><span style="min-height: 9pt; font-size: 9pt; color: rgb(1, 1, 1); font-family: &quot;Lucida Console&quot;;">Sequence unavailable</span></div><div style="min-height: 9pt; text-align: left;"></div><ul><li><div style="min-height: 9pt; text-align: left;">Ran on plasmodium - still timed out</div></li></ul><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">Running MEME 2.1</div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="text-align: left;">meme 6.txt -dna -oc . -nostatus -time 60000 -mod anr -nmotifs 15 -minw 6 -maxw 20 -objfun de -neg 6control.txt -revcomp -markov_order 1 </div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">CHANGED THE NAMES OF THE GENES BY TAKING OUT SPECIES NAME with this command</div><div style="min-height: 9pt; text-align: left;">cat 1.txt | sed &quot;s/caenorhabditis_elegans_prjna13758|//&quot; &gt; 1_1.txt</div><div style="min-height: 9pt; text-align: left;"><a href="https://stackoverflow.com/questions/16623835/remove-a-fixed-prefix-suffix-from-a-string-in-bash">https://stackoverflow.com/questions/16623835/remove-a-fixed-prefix-suffix-from-a-string-in-bash</a></div><div style="min-height: 9pt; text-align: left;">Man sed </div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">Test Command: </div><div style="min-height: 9pt; text-align: left;">(grep -c &quot;WBGene&quot; 1_1.txt)</div><div style="min-height: 9pt; text-align: left;">1291</div><div style="min-height: 9pt; text-align: left;"> (grep -c &quot;WBGene&quot; 1.txt)</div><div style="min-height: 9pt; text-align: left;">1291</div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">#Bring back new directory wt</div><div style="min-height: 9pt; text-align: left;"># r is recursive as always for a directory</div><div style="min-height: 9pt; text-align: left;">scp -r  Elegans_MEME_clean <a href="mailto:rachel.woo@plasmodium.eeb.utoronto.ca">rachel.woo@plasmodium.eeb.utoronto.ca</a>:~/meme1/<br/></div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">RUN TESTS WITH SHORTEST AND LONGEST FILES: </div><div style="min-height: 9pt; text-align: left;">Shortest == 6</div><div style="min-height: 9pt; text-align: left;">Longest == 12</div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">meme 6_1.txt -dna -oc . -nostatus -time 60000 -mod anr -nmotifs 15 -minw 6 -maxw 20 -objfun de -neg 6control.txt -revcomp -markov_order 1 &amp;<br/></div><div style="min-height: 9pt; text-align: left;"></div><div style="min-height: 9pt; text-align: left;">meme 12_1.txt -dna -oc . -nostatus -time 60000 -mod anr -nmotifs 15 -minw 6 -maxw 20 -objfun de -neg 12control.txt -revcomp -markov_order 1 &amp;<br/></div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">SEE HOW MANY MOTIS 12_1 gets in 60 000 s = 16H </div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;"><br/></div><div style="min-height: 9pt; text-align: left;">RECALL: <a href="https://kb.iu.edu/d/afnz">https://kb.iu.edu/d/afnz</a></div></span>
</div></body></html> 